src/lib.rs

```diff
@@ -214,6 +217,50 @@ impl Client {
         self.get_multi(keys).await
     }
 
+    // Gets the given key with additional metadata.
+    ///
+    /// If the key is found, `Some(Value)` is returned, describing the metadata and data of the key.
+    ///
+    /// Otherwise, `None` is returned.
+    ///
+    /// Supported meta flags:
+    /// - h: return whether item has been hit before as a 0 or 1
+    /// - l: return time since item was last accessed in seconds
+    /// - t: return item TTL remaining in seconds (-1 for unlimited)
+    pub async fn meta_get<K: AsRef<[u8]>>(
+        &mut self,
+        key: K,
+        meta_flags: Option<&[&str]>,
+    ) -> Result<Option<Value>, Error> {
+        self.conn.write_all(b"mg ").await?;
+        self.conn
+            .write_all(Self::validate_key_length(key.as_ref())?)
```

Comment:
Should we validate the key length _before_ we write `mg` into the buffer? We should fail as early as possible if the request is invalid